Libraries/Components/StatusBar/StatusBar.js

```diff
@@ -185,6 +264,60 @@ class StatusBar extends React.Component<{
    */
   static currentHeight = StatusBarManager.HEIGHT;
 
+  /**
+   * Get the current frame (height and y offset) of the Status Bar.
+   *
+   * Returns a Promise that resolves to a frame object with a `y` and `height`.
+   */
+  static getCurrentFrame(): Promise<{y: number, height: number}> {
+    return StatusBarManager.getCurrentFrame();
+  }
+
+  /**
+   * Adds an event handler. Supported events:
+   *
+   * - `statusBarFrameDidChange`: Fires when the status bar frame has changed.
+   *   The argument to the event handler is a frame object with the keys:
+   *   - `height`: The current height of the status bar.
+   *   - `y`: The y offset of the status bar, if any. This generally only happens
+   *     when the iOS In-Call Status Bar is visible and pushes the app down
+   *     instead of changing the size.
+   */
+  static addEventListener(
+    eventName: EventName,
+    handler: Function
+  ): {remove: () => void} {
+    if (eventName !== FRAME_DID_CHANGE_EVENT) {
```

Comment:
This is an error check of the arguments to make sure they actually follow the flow spec that makes this supposedly "unreachable".

Do I have to do something to avoid this incorrect warning?

Reply:
We don't need this, because it enforced by Flow.
Just move this condition into `invariant`.